code/test/VsTemplates.Test/VsTemplatesContentValidationTests.cs

```diff
@@ -0,0 +1,160 @@
+        };
+
+        public static IEnumerable<object[]> GetWizardProjectTemplateFiles()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            foreach (var path in WizardProjectTemplatesRoot)
+            {
+                var filePaths = GetFilesFromPath(path);
+                vsTemplatesPaths.AddRange(filePaths.Where(p => Path.GetExtension(p) == VsTemplateFileExtension)
+                    .Select(p => new object[] { p }));
+            }
+
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsTemplatePaths()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            var vsItemTemplatesPaths = GetAllVsTemplatePaths(ItemTemplatesRoot).Select(p => new object[] { p });
+            var vsProjectTemplatesPaths = GetAllVsTemplatePaths(ProjectTemplatesRoot).Select(p => new object[] { p });
+            vsTemplatesPaths.AddRange(vsItemTemplatesPaths);
+            vsTemplatesPaths.AddRange(vsProjectTemplatesPaths);
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsTemplateFiles()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            var vsItemTemplatesPaths = GetAllVsItemTemplateFiles();
+            var vsProjectTemplatesPaths = GetAllVsProjectTemplateFiles();
+            vsTemplatesPaths.AddRange(vsItemTemplatesPaths);
+            vsTemplatesPaths.AddRange(vsProjectTemplatesPaths);
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsProjectTemplateFiles()
+        {
+            var filePaths = GetFilesFromPath(ProjectTemplatesRoot);
+            var vsProjectTemplatesPaths = filePaths.Where(p => Path.GetExtension(p) == VsTemplateFileExtension);
+            return vsProjectTemplatesPaths.Select(p => new object[] { p });
+        }
+
+        private static IEnumerable<object[]> GetAllVsItemTemplateFiles()
+        {
+            var filePaths = GetFilesFromPath(ItemTemplatesRoot);
+            var vsItemTemplatesPaths = filePaths.Where(p => Path.GetExtension(p) == VsTemplateFileExtension);
+            return vsItemTemplatesPaths.Select(p => new object[] { p });
+        }
+
+        private static IEnumerable<string> GetAllVsTemplatePaths(string directory)
+        {
+            foreach (var dir in Directory.GetDirectories(directory))
+            {
+                if (Directory.GetFiles(dir).Any(f => Path.GetExtension(f) == VsTemplateFileExtension))
+                {
+                    yield return dir;
+                }
+                else
+                {
+                    foreach (var subDir in GetAllVsTemplatePaths(dir))
+                    {
+                        yield return subDir;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetFilesFromPath(string directory)
```

Comment:
Why the need to manually recurse through directories? Can't you use the version of Directory.GetFiles that will do this for you?